### DISCUSSION TITLE
Why SWC is faster but have smaller compile results than Terser + Babel?

### DISCUSSION THREAD

**Discussion by LikeDreamwalker** [swc-project/swc#8497] [General]
I am trying to change my team's Minify plugin to SWC, and we are excited about it.
But here is one thing, we can understand SWC is faster because it can have a good use of multiple cores and threads; but why the compile results' size is also smaller?
I have compared some of the compiled files and found out that SWC seems like it will make the code more concise, but does it have some special and advanced strategy than Terser?
I haven't found any documents or blogs that can explain this. I hope somebody can bring me more info about the SWC's Minify strategy.

Also, I have found a benchmark that can prove that SWC has smaller Minify results:
https://github.com/privatenumber/minification-benchmarks

**Comment by kwonoj** [comment]
SWC applies various optimization strategies in combination, brings  better outcome for ceratin usecases. We do not have a detailed explanation for each, the best bet is probably reading source codes.

**Reply by LikeDreamwalker** [reply]
Thanks for your reply, and could you please tell me where the Minify core's code is located? New for SWC and Rust.
